sovryn-token-bridge/bridge/contracts/Bridge.sol

```diff
@@ -207,14 +212,18 @@ contract Bridge is Initializable, IBridge, IERC777Recipient, UpgradablePausable,
         //// Bridge v4 upgrade functions
         if (tokenAddress == WETHAddr) {
             address payable payableReceiver = address(uint160(receiver));
-            (bool success, ) =
-                payableReceiver.call.value(amount)("");
-            require(success, "Bridge: Failed to send ETH to receiver");
+            if (isBridgeReceiver[receiver]) {
+                IBridgeReceiver(payableReceiver).receiveEthFromBridge.value(amount)(userData);
+            } else {
+                (bool success, ) =
+                    payableReceiver.call.value(amount)("");
+                require(success, "Bridge: Failed to send ETH to receiver");
+            }
         }
         else {
             IERC20(tokenAddress).safeTransfer(receiver, formattedAmount);
         }
-        emit AcceptedCrossTransfer(tokenAddress, receiver, amount, decimals, granularity, formattedAmount, calculatedDecimals, 1, "");
+        emit AcceptedCrossTransfer(tokenAddress, receiver, amount, decimals, granularity, formattedAmount, calculatedDecimals, 1, userData);
```

Comment:
Emitting an event with non-empty `userData` seems inconsistent with an actual call in a not-BridgeReceiver target case.

Reply:
This is true, but userData is NA in this case

Reply:
evengy is right. there is the possibility that i am adding user data, but the receiver is not the bridge. 
in that case the event would be emitting user data, although it was not passed on. we can still have it this way, but the function documentation should then clearly explain that userData is merely used for logging in case that the receiver is not the bridge.